test/hotspot/jtreg/compiler/escapeAnalysis/TestReduceAllocationAndNestedScalarized.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8331194
+ * @summary Check that Reduce Allocation Merges doesn't crash when an input
+ *          of the Phi is not the _current_ output of the Phi but said input
+ *          needs to be rematerialized because it's used regardless of the
+ *          Phi output.
+ * @requires vm.compiler2.enabled
```

Comment:
Again, I think you do not neet do restrict test execution to C2. The `requires` does not make C2 appear magically, but rather it just means the test is not executed if C2 is not available. So why restrict the test? Maybe some other compiler also has a similar or totally different bug triggered with your test. Would that not be nice?

Reply:
Now that you have `-XX:+IgnoreUnrecognizedVMOptions`, you do not have to be worried that some other compiler crashes because of some unrecognized flag.

Reply:
> Would that not be nice?

Fingers crossed!